### PR TITLE
Snack chat page

### PR DESCRIPTION
**Pull request by HelloUniverseNo31415**
# Checklist:

Related work items: #72 

Preview url: https://honesty-store-kiosk-dev.firebaseapp.com/snackChat

- [ ] Tests have been added for new functionality [N/A]
- [x] Work has been reviewed by other team members
- [ ] Work has been reviewed by @craigayre or @eclennett

### CODE REVIEW

File: src/components/OldApp/OldApp.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import WebcamCaptureContainer from '../WebcamCapture/WebcamCaptureContainer.js';
+import WebcamCaptureContainer from '../WebcamCapture/container';
```
Comment (cainy393) [inline src/components/OldApp/OldApp.js]: How come this file name has reverted back?

File: src/components/SnackChat/SnackChat.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, {Component} from 'react';
+import WebcamCapture from '../WebcamCapture/container';
```
Comment (cainy393) [inline src/components/SnackChat/SnackChat.js]: And here too

File: src/components/OldApp/OldApp.js (hunk 1 above)
Comment (HelloUniverseNo31415) [reply src/components/OldApp/OldApp.js]: My guess? witchcraft!

File: src/components/ConfirmationBox/ConfirmationBox.js [hunk 3]
```diff
@@ -4,7 +4,8 @@ import PropTypes from 'prop-types';
 class ConfirmationBox extends React.Component {
 	handleYes = () => {
 		this.props.setActualItem(this.props.name);
-		this.props.history.push('/slackname');
+		const nextPage = this.props.sendWithPhoto ? 'snackchat' : 'slackname';
```
Comment (paulhands) [inline src/components/ConfirmationBox/ConfirmationBox.js]: This is probably one of those places where destructuring might look nicer
e.g. add a line

```const { setActualItem, name, sendWithPhoto, history } = this.props;```

then remove all the ```this.props``` from elsewhere

File: src/components/SnackChat/SnackChat.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, {Component} from 'react';
+import WebcamCapture from '../WebcamCapture/WebcamCaptureContainer';
+import PropTypes from 'prop-types';
+
+class SnackChat extends Component {
+	initialTime = new Date();
+	state = {counter: 5};
```
Comment (paulhands) [inline src/components/SnackChat/SnackChat.js]: if you want to use internal state in a Component you need a constructor and an initial state, it's a bit more complex than just defining state = {counter: 5}

```
constructor(props) {
    super(props);
    this.state = {counter: 5};
  }
```

inside the component class. You shouldn't modify the state directly.

https://reactjs.org/docs/state-and-lifecycle.html

File: src/components/SnackChat/SnackChatReducer.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import {SET_SNACK_CHAT} from './SnackChatActionTypes';
+
+export function snackChat(state = {}, action) {
+	console.log(action.type);
+	console.log(action.snackChat);
```
Comment (paulhands) [inline src/components/SnackChat/SnackChatReducer.js]: these should be removed

File: src/components/SnackChat/SnackChat.js (hunk 4 above)
Comment (HelloUniverseNo31415) [reply src/components/SnackChat/SnackChat.js]: So I have set the initial state outside of the constructor before, is this a thing I shouldn't do at all or is this a special case?

Comment (paulhands) [reply src/components/SnackChat/SnackChat.js]: Should avoid doing, but correct it in a different PR

Comment (paulhands) [reply src/components/SnackChat/SnackChat.js]: (Obviously correct this instance now)

File: src/components/SnackChat/SnackChat.js [hunk 6]
```diff
@@ -4,7 +4,10 @@ import PropTypes from 'prop-types';
 
 class SnackChat extends Component {
 	initialTime = new Date();
-	state = {counter: 5};
+	constructor(props) {
+    super(props);
```
Comment (paulhands) [inline src/components/SnackChat/SnackChat.js]: tabbing looks off here?

**Pull request merged**